python-package/xgboost/spark/core.py

```diff
@@ -0,0 +1,810 @@
+import shutil
+import tempfile
+from typing import Iterator, Tuple
+import numpy as np
+import pandas as pd
+from scipy.special import expit, softmax
+from pyspark.ml import Estimator, Model
+from pyspark.ml.param.shared import HasFeaturesCol, HasLabelCol, HasWeightCol, \
+    HasPredictionCol, HasProbabilityCol, HasRawPredictionCol, HasValidationIndicatorCol
+from pyspark.ml.param import Param, Params, TypeConverters
+from pyspark.ml.util import MLReadable, MLWritable
+from pyspark.sql.functions import col, pandas_udf, countDistinct, struct
+from pyspark.sql.types import ArrayType, FloatType
+from xgboost import XGBClassifier, XGBRegressor
```

Comment:
@trivialfis 
In PySpark-XGBoost integration, the historical code we reuse some API from `xgboost.sklearn` (e.g. the XGBClassifier, XGBRegressor),
on second thought, I think it is better to make the implementation only rely on the xgboost core API, what do you think ?